Terminal.Gui/ConsoleDrivers/ConsoleDriver.cs

```diff
@@ -32,15 +32,35 @@ public abstract class ConsoleDriver {
 	/// </summary>
 	internal static bool RunningUnitTests { get; set; }
 
+	#region Setup & Teardown
+
+	/// <summary>
+	/// Returns an instance of <see cref="MainLoop"/> using the <see cref="IMainLoopDriver"/> for the driver.
+	/// </summary>
+	/// <returns></returns>
+	internal abstract MainLoop CreateMainLoop ();
+
+	/// <summary>
+	/// Initializes the driver
+	/// </summary>
+	/// <param name="terminalResized">Method to invoke when the terminal is resized.</param>
+	internal abstract void Init (Action terminalResized);
```

Comment:
In the history of gui.cs, nobody but the maintainers (us) has created a ConsoleDriver or IMainLoop implementation. 

I am asserting that nobody ever will in the future either. 

If I'm wrong and someday someone says "I have a new ConsoleDriver I'd like to write" then we can help them do that as part of the library. This will be far simpler than exposing a bunch of fragile public APIs that may or may not work anyway.

Note that in a few minutes I'll be pushing an update to this PR that goes even further. `Init` has NO parameters and `PrepareToRun` will be really close to being able to be deleted as well... ;-)

Reply:
> In the history of gui.cs, nobody but the maintainers (us) has created a ConsoleDriver or IMainLoop implementation.
> 
Here is one https://github.com/Blazor-Console/HACC.

> I am asserting that nobody ever will in the future either.
> 
We never know :-)

> If I'm wrong and someday someone says "I have a new ConsoleDriver I'd like to write" then we can help them do that as part of the library. This will be far simpler than exposing a bunch of fragile public APIs that may or may not work anyway.
> 
Ok I see it only must done inside the API.

> Note that in a few minutes I'll be pushing an update to this PR that goes even further. `Init` has NO parameters and `PrepareToRun` will be really close to being able to be deleted as well... ;-)

Wow that rocks :-)

Reply:
> > In the history of gui.cs, nobody but the maintainers (us) has created a ConsoleDriver or IMainLoop implementation.
> 
> Here is one https://github.com/Blazor-Console/HACC.
> 
> > I am asserting that nobody ever will in the future either.
> 
> We never know :-)
> 
> > If I'm wrong and someday someone says "I have a new ConsoleDriver I'd like to write" then we can help them do that as part of the library. This will be far simpler than exposing a bunch of fragile public APIs that may or may not work anyway.

Ok, I was wrong about there not being any examples. Thanks for pointing this out.

We have several choices:

1) Make it so arbitrary ConsoleDrivers could be developed outside of the library (revert to v1 patterns)
2) Extend the v2 pattern in this PR to enable JUST the what is needed for HACC to work.
3) Fold the requirements of HACC into the libary, creating a new built-in `WebConsoleDriver`

I'm not a fan of #1, but #2 or #3 seem very doable.